frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -609,96 +643,120 @@ export const surveyLogic = kea<surveyLogicType>([
             },
         },
     })),
-    listeners(({ actions, values }) => ({
-        createSurveySuccess: ({ survey }) => {
-            lemonToast.success(<>Survey {survey.name} created</>)
-            actions.loadSurveys()
-            router.actions.replace(urls.survey(survey.id))
-            actions.reportSurveyCreated(survey)
-        },
-        updateSurveySuccess: ({ survey }) => {
-            lemonToast.success(<>Survey {survey.name} updated</>)
-            actions.editingSurvey(false)
-            actions.reportSurveyEdited(survey)
-            actions.loadSurveys()
-        },
-        duplicateSurveySuccess: () => {
-            actions.loadSurveys()
-        },
-        launchSurveySuccess: ({ survey }) => {
-            lemonToast.success(<>Survey {survey.name} launched</>)
-            actions.loadSurveys()
-        },
-        stopSurveySuccess: () => {
-            actions.loadSurveys()
-        },
-        resumeSurveySuccess: () => {
-            actions.loadSurveys()
-        },
-        archiveSurvey: () => {
-            actions.updateSurvey({ archived: true })
-        },
-        loadSurveySuccess: () => {
+    listeners(({ actions, values }) => {
```

Comment:
miss leading number of lines changed here.

since I added this new function, `reloadAllSurveysResults`, it shifted a lot of things down.

but the main changes are using `switch` case instead of multiple ifs and elses, and using this new function on both `setPropertyFilters` and `setAnswersFilters`